### PR TITLE
ups canonical and abstract types

### PR DESCRIPTION
**Pull request by adelin-diac**

### CODE REVIEW

File: ontology/yaml/resources/ELECTRICAL/entity_types/UPS.yaml [hunk 1]
```diff
@@ -74,3 +74,11 @@ UPS_PWM:
   implements:
   - UPS
   - PWM
+
+UPS_BBM_UPSM:
```
Comment (shambergoldstein) [inline ontology/yaml/resources/ELECTRICAL/entity_types/UPS.yaml:78]: Based on above comments, should this instead be `UPS_SS_UPSBM`

**Review by shambergoldstein — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/ELECTRICAL/entity_types/UPS.yaml (hunk 1 above)
Comment (adelin-diac) [reply ontology/yaml/resources/ELECTRICAL/entity_types/UPS.yaml:78]: Thanks for the feedback on this, really cleared up a lot of concerns I had. I've updated the PR based on your comments.

**Review by cstirdivant — APPROVED** [review]

File: ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml [hunk 2]
```diff
@@ -23,6 +23,7 @@ SS:
   - failed_alarm
   - lost_power_alarm
   - power_status
+  - bypass_status
```
Comment (cstirdivant) [inline ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml:26]: Apologies for the confusion; please remove here and add the `bypass_status` field as an `opt_uses` on the `UPS` in the GENERALTYPES.yaml

**Review by cstirdivant — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml (hunk 2 above)
Comment (adelin-diac) [reply ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml]: updated

**Review by cstirdivant — APPROVED** [review]

**Pull request merged**